src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -1069,12 +1137,39 @@ internal static void ByteToHexDigits(Span<char> outChars, byte val)
             outChars[1] = BinaryToHexDigit(val & 0xF);
         }
 
-        #endregion
+        internal static bool IsLowerCaseHexAndNotAllZeros(ReadOnlySpan<char> idData)
+        {
+            // Verify lower-case hex and not all zeros https://w3c.github.io/trace-context/#field-value
+            bool isNonZero = false;
+            int i = 0;
+            for (; i < idData.Length; i++)
+            {
+                char c = idData[i];
+                if (!IsHexadecimalChar(c))
+                {
+                    break;
+                }
 
-        readonly ulong _id1;
-        readonly ulong _id2;
-        readonly string _asHexString;  // Caches the Hex string    
-        #endregion
+                if (c != '0')
+                {
+                    isNonZero = true;
+                }
+            }
+
+            if (i < idData.Length || !isNonZero)
+            {
+                return false;
+            }
+
+            return true;
```

Comment:
```C#
return i == idData.Length && isNonZero;
```
?

Reply:
Or can you just `return false;` from where you `break;` and then just make this `return isNonZero;`?

Reply:
Resolved